mav_trajectory_generation_ros/include/mav_trajectory_generation_ros/feasibility_base.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2016, Markus Achtelik, ASL, ETH Zurich, Switzerland
+ * Copyright (c) 2016, Michael Burri, ASL, ETH Zurich, Switzerland
+ * Copyright (c) 2016, Helen Oleynikova, ASL, ETH Zurich, Switzerland
+ * Copyright (c) 2016, Rik Bähnemann, ASL, ETH Zurich, Switzerland
+ * Copyright (c) 2016, Marija Popovic, ASL, ETH Zurich, Switzerland
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MAV_TRAJECTORY_GENERATION_ROS_FEASIBILITY_BASE_H_
+#define MAV_TRAJECTORY_GENERATION_ROS_FEASIBILITY_BASE_H_
+
+#include <glog/logging.h>
+#include <ros/ros.h>
+#include <Eigen/Core>
+#include <Eigen/StdVector>
+
+#include <mav_trajectory_generation/trajectory.h>
+
+namespace mav_trajectory_generation {
+enum InputFeasibilityResult {
+  kInputFeasible = 0,    // The trajectory is input feasible.
+  kInputIndeterminable,  // Cannot determine whether the trajectory is feasible
+                         // with respect to the inputs.
+  kInputInfeasibleThrustHigh,      // The trajectory is infeasible, failed max.
+                                   // thrust test first.
+  kInputInfeasibleThrustLow,       // The trajectory is infeasible, failed min.
+                                   // thrust test first.
+  kInputInfeasibleVelocity,        // The Trajectory is infeasible, failed max.
+                                   // velocity test first.
+  kInputInfeasibleRollPitchRates,  // The trajectory is infeasible, failed max.
+                                   // roll/pitch rates test first.
+  kInputInfeasibleYawRates,  // The trajectory is infeasible, faild max. yaw
+                             // rates test first.
+  kInputInfeasibleYawAcc,    // The trajectory is infeasible, failed max. yaw
+                             // acceleration test first.
+};
+
+// Human readable InputFeasibilityResult.
+std::string getInputFeasibilityResultName(InputFeasibilityResult fr);
+
+// Dynamic constraints of the MAV.
+class InputConstraints {
```

Comment:
My only question here is should we be able to turn on/turn off constraints at will? I can see cases where you'd want to check just a few, not all?